cmd/socksclient/socksclient.go

```diff
@@ -3,39 +3,48 @@ package main
 import (
 	"flag"
 	"fmt"
-	"io/ioutil"
 	"net"
 	"net/http"
+	"net/http/httputil"
 	"net/url"
 	"os"
 	"time"
 
 	"go.uber.org/zap"
 )
 
-var logger *zap.Logger
-
-func init() {
-	l, _ := zap.NewProduction()
-	logger = l
-}
-
 func main() {
-	defer logger.Sync()
-	flagSocksServer := flag.String("socks-server", "", "addr of socks server like socks://user:pass@127.0.0.1:8000")
+	log, _ := zap.NewProduction()
+	defer log.Sync()
+	flagSocksServer := flag.String("socks-server", "socks5://test:test@127.0.0.1:8000", "addr of socks server like socks://user:pass@127.0.0.1:8000")
 	flag.Parse()
 
 	if len(flag.Args()) != 1 {
-		logger.Fatal("usage: " + os.Args[0] + " -socks-server=127.0.0.1:8080 http://www.google.com/")
+		log.Fatal("usage: " + os.Args[0] + " -socks-server=127.0.0.1:8000 http://www.google.com/")
 	}
 
 	urlToFetch := flag.Arg(0)
 
 	proxyURL, errProxyURL := url.Parse(*flagSocksServer)
 	if errProxyURL != nil {
-		logger.Fatal("invalid proxy server:", zap.Error(errProxyURL))
+		log.Fatal("invalid proxy server:", zap.Error(errProxyURL))
 	}
 
+	response, err := newClient(proxyURL).Get(urlToFetch)
+	if err != nil {
+		log.Fatal("could not GET", zap.Error(err), zap.String("url", urlToFetch))
+	}
+	defer response.Body.Close()
+
+	respBytes, err := httputil.DumpResponse(response, true)
+	if err != nil {
+		log.Fatal("failed httputil.DumpResponse", zap.Error(err))
+	}
+
+	fmt.Println(string(respBytes))
```

Comment:
Why not log?

Reply:
when you see it :-) log has ugly output and this client is a debug client